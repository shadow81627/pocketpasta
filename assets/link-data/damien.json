{
  "@context": "http://schema.org",
  "@type": "Person",
  "description": "Full stack web developer at XCOM Media, Griffith University graduate. Aspiring entrepreneur with a passion for startup weekends and hackathons.",
  "keywords": "html, css, vue, bootstrap",
  "email": "mailto:damien.robinson@pocketpasta.com",
  "jobTitle": "Full Stack Web Developer",
  "brand": "PocketPasta",
  "name": "Damien Robinson",
  "givenName": "Damien",
  "familyName": "Robinson",
  "alumniOf": "Griffith University",
  "gender": "male",
  "nationality": "Australian",
  "url": "https://pocketpasta.com",
  "sameAs": [
    "https://github.com/shadow81627",
    "https://www.linkedin.com/in/damien-robinson-788925101/",
    "https://twitter.com/Shadow81627"
  ]
}
