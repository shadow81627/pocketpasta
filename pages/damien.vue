<template>
  <div class="container">
    <div class="row">
      <profile />
    </div>
  </div>
</template>

<script>
// @ is an alias to /src
import damien from '@/assets/link-data/damien.json';

import Profile from '@/components/Profile.vue';

export default {
  components: {
    Profile,
  },
  data: () => ({
    damien,
  }),
  head() {
    return {
      script: [
        {
          json: damien,
          type: 'application/ld+json',
        },
      ],
    };
  },
};
</script>
