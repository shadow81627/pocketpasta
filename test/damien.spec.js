import damien from '@/pages/damien.vue';
import { shallowMount } from '@vue/test-utils';

const factory = () => shallowMount(damien, {});

describe('damien', () => {
  test('mounts properly', () => {
    const wrapper = factory();
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test('renders properly', () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });
  test('head', () => {
    expect(damien.head()).toBeTruthy();
  });
});
