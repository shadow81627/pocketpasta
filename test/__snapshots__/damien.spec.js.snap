// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`damien renders properly 1`] = `
"<div class=\\"container\\">
  <div class=\\"row\\">
    <profile-stub></profile-stub>
  </div>
</div>"
`;
